File: onebot/src/main/java/cn/evole/onebot/sdk/response/group/GroupFilesResp.java
package cn.evole.onebot.sdk.response.group;

import cn.evole.onebot.sdk.util.json.GroupFilesAdapter;
import com.google.gson.annotations.JsonAdapter;
import com.google.gson.annotations.SerializedName;
import lombok.Data;

import java.util.List;

/**
 * @author cnlimiter
 */
@Data
public class GroupFilesResp {

    public List<Files> files;

    public List<Folders> folders;

    /**
     * 群文件
     */
    @Data
    @JsonAdapter(GroupFilesAdapter.Files.class)
    public static class Files {
        public String fileId;
        public String fileName;
        public int busid;
        public long fileSize;
        public long uploadTime;
        public long deadTime;
        public long modifyTime;
        public int downloadTimes;
        public long uploader;
        public String uploaderName;
        public String md5 = null;
        public String sha1 = null;
        public String sha3 = null;
    }

    /**
     * 群文件夹
     */
    @Data
    public static class Folders {

        @SerializedName("folder_id")
        public String folderId;

        @SerializedName("folder_name")
        public String folderName;

        @SerializedName("create_time")
        public long createTime;

        public long creator;

        @SerializedName("creator_name")
        public String creatorName;

        @SerializedName("total_file_count")
        public int totalFileCount;

    }

}

File: onebot/src/main/kotlin/sdk/response/group/GroupFilesResp.kt
package cn.evolvefield.onebot.sdk.response.group

import cn.evolvefield.onebot.sdk.util.json.GroupFilesAdapter
import com.google.gson.annotations.JsonAdapter
import com.google.gson.annotations.SerializedName
import lombok.Data

@Data
class GroupFilesResp {

    var files: List<Files>? = null

    var folders: List<Folders>? = null

    /**
     * 群文件
     */
    @Data
    @JsonAdapter(GroupFilesAdapter.Files::class)
    class Files {
        var fileId = ""
        var fileName = ""
        var busid = 0
        var fileSize = 0L
        var uploadTime = 0L
        var deadTime = 0L
        var modifyTime = 0L
        var downloadTimes = 0
        var uploader = 0L
        var uploaderName = ""
        var md5: String? = null
        var sha1: String? = null
        var sha3: String? = null
    }

    /**
     * 群文件夹
     */
    @Data
    class Folders {
        @SerializedName("folder_id")
        var folderId = ""
        @SerializedName("folder_name")
        var folderName = ""
        @SerializedName("create_time")
        var createTime = 0L
        @SerializedName("creator")
        var creator = 0L
        @SerializedName("creator_name")
        var creatorName = ""
        @SerializedName("total_file_count")
        var totalFileCount = 0
    }
}
